diff --git a/CMakeLists.txt b/CMakeLists.txt
index a692bb1..7a3b433 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,8 +84,6 @@ install( TARGETS qtgreet DESTINATION bin )
 install( DIRECTORY backgrounds DESTINATION share/qtgreet )
 
 # Configuration
-install( FILES configs/config.ini DESTINATION /etc/qtgreet )
-install( FILES configs/wayfire.ini DESTINATION /etc/qtgreet )
 
 # Data files
 install( FILES README.md Changelog ReleaseNotes DESTINATION share/qtgreet )
